src/sentry/tasks/reports.py

```diff
@@ -270,16 +319,48 @@ def prepare_project_issue_summaries(interval, project):
     return [new_issue_count, reopened_issue_count, existing_issue_count]
 
 
-def prepare_project_usage_summary(start__stop, project):
+def prepare_project_usage_outcomes(start__stop, project):
     start, stop = start__stop
-    return (
-        tsdb.get_sums(
-            tsdb.models.project_total_blacklisted, [project.id], start, stop, rollup=60 * 60 * 24
-        )[project.id],
-        tsdb.get_sums(
-            tsdb.models.project_total_rejected, [project.id], start, stop, rollup=60 * 60 * 24
-        )[project.id],
+
+    # XXX(epurkhiser): Tsdb used to use day buckets, where the end would
+    # represent a whole day. Snuba queries more accurately thus we must
+    # capture the entire last day
+    end = stop + timedelta(days=1)
+
+    query = Query(
```

Comment:
LGTM.

Reply:
Did you consider using `sentry.snuba.outcomes.run_outcomes_query()` here? Might be good to have reports use similar logic/query building that we use for the API endpoints.

Reply:
I actually did, but I decided against it for a few reasons:

1) `run_outcomes_query()` makes two queries, one for the sums and another for the time series, I could probably split this out  https://github.com/getsentry/sentry/blob/0fca29ef4dee644e4f1df5f1469305af71c7e0f2/src/sentry/snuba/outcomes.py#L271-L294
  
2) The `QueryDescriptor` would need some refactoring (which I did start in https://github.com/getsentry/sentry/pull/25289 

3) It didn't really seem to need any of the logic that we have to row translation